backend-rust/src/indexer.rs

```diff
@@ -918,6 +923,11 @@ impl PreparedBlock {
         )
         .await?;
         let baker_unmark_suspended = PreparedUnmarkPrimedForSuspension::prepare(data)?;
+
+        let election_info =
+            node_client.get_election_info(data.finalized_block_info.height).await?.response;
```

Comment:
This query will return the same information for every block within the same reward period, so querying it at each payday block would be enough

Reply:
Good point, I initial thought the `lottery_power` is computed by the current stake at the queried `block_height` of the baker in the `get_election_info(block_height)`, but it seems to return the historical data from the last payday_block:

https://github.com/Concordium/concordium-node/blob/5dcc2b4ca777d65f119871c2735c01d410bc1f09/concordium-consensus/src/Concordium/Queries.hs#L1092